src/compiler/crystal/semantic/restrictions.cr

```diff
@@ -281,10 +281,10 @@ module Crystal
 
   class Generic
     def restriction_of?(other : Path, owner)
-      other_type = owner.lookup_type?(self)
-      if other_type
-        self_type = owner.lookup_path(other)
-        if self_type
+      self_type = owner.lookup_type?(self)
+      if self_type
+        other_type = owner.lookup_path(other)
+        if other_type
```

Comment:
@asterite you'd have to explain to my boss why I didn't slept last night :joy: :innocent:

Reply:
Wow! How did that even work? 😮